server/bin/www

```diff
@@ -20,22 +23,42 @@ connectDB();
  * Get port from environment and store in Express.
  */
 
-var port = normalizePort(process.env.PORT || "3001");
-app.set("port", port);
+var port = normalizePort(process.env.PORT || '3001');
+app.set('port', port);
 
 /**
  * Create HTTP server.
  */
-
 var server = http.createServer(app);
 
+/**
+ * Socket Connection
+ */
+var io = socketio(server);
+io.use(async (socket, next) => {
+  try {
+    const token = socket.handshake.query.token;
```

Comment:
Let's try using cookies actually. Have you tried `socket.handshake.headers.cookie)`?
`